ocd-2159.sql

```diff
@@ -0,0 +1,158 @@
+	CONSTRAINT change_request_status_type_pk PRIMARY KEY (id)
+);
+CREATE TRIGGER change_request_status_type_audit AFTER INSERT OR UPDATE OR DELETE on openchpl.change_request_status_type FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();
+CREATE TRIGGER change_request_status_type_timestamp BEFORE UPDATE on openchpl.change_request_status_type FOR EACH ROW EXECUTE PROCEDURE openchpl.update_last_modified_date_column();
+
+CREATE TABLE openchpl.change_request (
+    id bigserial NOT NULL,
+    change_request_type_id bigint NOT NULL,
+    developer_id bigint NOT NULL,
+    creation_date timestamp NOT NULL DEFAULT NOW(),
+	last_modified_date timestamp NOT NULL DEFAULT NOW(),
+	last_modified_user bigint NOT NULL,
+	deleted bool NOT NULL DEFAULT false,
+	CONSTRAINT change_request_pk PRIMARY KEY (id),
+    CONSTRAINT change_request_type_fk FOREIGN KEY (change_request_type_id)
+	    REFERENCES openchpl.change_request_type (id)
+        MATCH SIMPLE ON UPDATE NO ACTION ON DELETE RESTRICT,
+    CONSTRAINT developer_fk FOREIGN KEY (developer_id)
+        REFERENCES openchpl.vendor (vendor_id)
+        MATCH SIMPLE ON UPDATE NO ACTION ON DELETE RESTRICT
+);
+CREATE TRIGGER change_request_audit AFTER INSERT OR UPDATE OR DELETE on openchpl.change_request FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();
+CREATE TRIGGER change_request_timestamp BEFORE UPDATE on openchpl.change_request FOR EACH ROW EXECUTE PROCEDURE openchpl.update_last_modified_date_column();
+
+CREATE TABLE openchpl.change_request_status (
+    id bigserial NOT NULL,
+    change_request_id bigint NOT NULL,
+    change_request_status_type_id bigint NOT NULL,
+    status_change_date timestamp NOT NULL,
+    comment text,
+    certification_body_id bigint,
+    creation_date timestamp NOT NULL DEFAULT NOW(),
+	last_modified_date timestamp NOT NULL DEFAULT NOW(),
+	last_modified_user bigint NOT NULL,
+	deleted bool NOT NULL DEFAULT false,
+	CONSTRAINT change_request_status_pk PRIMARY KEY (id),
+    CONSTRAINT change_request_fk FOREIGN KEY (change_request_id)
+	    REFERENCES openchpl.change_request (id)
+        MATCH SIMPLE ON UPDATE NO ACTION ON DELETE RESTRICT,
+    CONSTRAINT change_request_status_type_fk FOREIGN KEY (change_request_status_type_id)
+	    REFERENCES openchpl.change_request_status_type (id)
+        MATCH SIMPLE ON UPDATE NO ACTION ON DELETE RESTRICT,
+    CONSTRAINT certification_body_fk FOREIGN KEY (certification_body_id)
+	    REFERENCES openchpl.certification_body (certification_body_id)
+        MATCH SIMPLE ON UPDATE NO ACTION ON DELETE RESTRICT
+);
+CREATE TRIGGER change_request_status_audit AFTER INSERT OR UPDATE OR DELETE on openchpl.change_request_status FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();
+CREATE TRIGGER change_request_status_timestamp BEFORE UPDATE on openchpl.change_request_status FOR EACH ROW EXECUTE PROCEDURE openchpl.update_last_modified_date_column();
+
+CREATE TABLE openchpl.change_request_certification_body_map (
```

Comment:
Just wondering about this table and how it's expected to be used... I went back and looked at the requirements in the Epic and see that the system should notify "Any ACB that the Developer has a Listing with" about change requests so I thought maybe for that? But the listing's ACB can be changed so I think we'd calculate "Any ACB that the developer has a listing with" on the fly in one of the permissions classes rather than save it here. Elsewhere in the tickets, the way things are written it's more like any ACB should be able to view/manage any request and in that case I don't think you'd need a mapping. So anyway, I just wanted to understand this table. I haven't dug through the 90+ API files changed yet to figure it out myself. :)

Reply:
I think this is for saving the "acting ACB" when a change is modified, though I'm only looking at the comment, so might be thinking of a different table

Reply:
Should it be joined with the change_request_status table then?

Reply:
Should it be joined with the change_request_status table then?

Reply:
Or actually the certification body id would just be a column in that status table then right?